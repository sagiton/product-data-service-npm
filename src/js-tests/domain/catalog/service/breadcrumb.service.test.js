import * as _ from 'lodash'

describe('service: BreadcrumbService', () => {
    let BreadcrumbService
    let catalogService;

    beforeEach(angular.mock.module('pds.catalog.service'))

    beforeEach(angular.mock.module($provide => {
        catalogService = {
            getTemplate: jest.fn(),
            resolveUriFromHierarchy: jest.fn()
        }
        $provide.value('CatalogService', catalogService)
        $provide.value('_', _)
        // $provide.value('$q', Promise)
    }))

    beforeEach(angular.mock.inject(($injector) => {
        BreadcrumbService = $injector.get('breadcrumbService')
    }));

    it('should return breadcrumbs', async (done) => {
        spyOn(catalogService, 'getTemplate').and.returnValue(Promise.resolve({
            nodes: [
                {
                    "id": 669445,
                    "type": "ROOT_CATEGORY",
                    "name": "Commercial & Industrial"
                },
                {
                    "id": 669446,
                    "type": "CATEGORY",
                    "name": "Produkte"
                }
            ]
        }))

        spyOn(catalogService, 'resolveUriFromHierarchy').and.returnValue(Promise.resolve('url.url'))

        BreadcrumbService
            .build()
            .then((breadcrumbs) => {
                expect(breadcrumbs).toMatchSnapshot()
            })
    })

})
