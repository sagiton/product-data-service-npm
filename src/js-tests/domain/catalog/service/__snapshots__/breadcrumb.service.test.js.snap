// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service: BreadcrumbService should return breadcrumbs 1`] = `
Array [
  Object {
    "id": 669445,
    "name": "Commercial & Industrial",
    "type": "ROOT_CATEGORY",
    "url": "url.url",
  },
  Object {
    "id": 669446,
    "name": "Produkte",
    "type": "CATEGORY",
    "url": "url.url",
  },
]
`;
